packages/embark-swarm/src/index.js

```diff
@@ -59,117 +38,187 @@ class Swarm {
         }
       });
       this.events.emit("swarm:process:started", null, false);
-      return callback();
+      return;
     }
 
-    this.providerUrl = buildUrl(this.config.storageConfig.upload.protocol, this.config.storageConfig.upload.host, this.config.storageConfig.upload.port);
+    this.embark.events.setCommandHandler("module:swarm:reset", (cb) => {
+      this.events.request("processes:stop", "swarm", (err) => {
+        if (err) {
+          this.logger.error(__('Error stopping Swarm process'), err);
+        }
+        this.init(cb);
+      });
+    });
+
+    this.events.request("runcode:whitelist", 'swarm-api', () => {});
+    this.events.request("runcode:whitelist", 'embarkjs', () => {});
+    this.events.request("runcode:whitelist", 'embarkjs-swarm', () => {});
+    this.events.on("storage:started", this.registerSwarmObject.bind(this));
+    this.events.on("storage:started", this.connectEmbarkJSProvider.bind(this));
 
-    this.getUrl = this.config.storageConfig.upload.getUrl || this.providerUrl + '/bzz:/';
+    let plugin = options.plugins.createPlugin('swarmplugin', {});
+    plugin.registerActionForEvent("pipeline:generateAll:before", this.addEmbarkJSSwarmArtifact.bind(this));
```

Comment:
No need to create a plugin. This is already plugin. You can do:
`embark.registerActionForEvent("pipeline:generateAll:before", this.addEmbarkJSSwarmArtifact.bind(this));`

So no need to pass `plugins` to this in options

Reply:
Thanks, fixed.